dev/build/windows/patches_coq/interval.patch

```diff
@@ -0,0 +1,187 @@
++  rewrite Hierarchy.plus_comm.
+   move: (H _ _ Qbu2 Qbv2) => /= .
+   move => /(_ _ HC) Hle.
+   apply: Rle_lt_trans Hle _.
+--- interval-839a03e1bddbafab868fbceee59abe678e32a0f3.orig/src/Interval/Eval.v	2019-07-20 09:48:54.000000000 +0200
++++ interval-839a03e1bddbafab868fbceee59abe678e32a0f3/src/Interval/Eval.v	2019-12-03 15:55:33.037225800 +0100
+@@ -62,8 +62,10 @@
+ now rewrite !IH; case no_floor_term; simpl; case no_floor_term; simpl.
+ Qed.
+ 
++Locate nil.
++
+ Lemma no_floor_prog_rcons t prog :
+-  no_floor_prog (prog ++ (t :: nil)) = no_floor_term t && no_floor_prog prog.
++  no_floor_prog (prog ++ (t :: List.nil)) = no_floor_term t && no_floor_prog prog.
+ Proof.
+ unfold no_floor_prog.
+ generalize true.
+--- interval-839a03e1bddbafab868fbceee59abe678e32a0f3.orig/src/Missing/Coquelicot.v	2019-07-20 09:48:54.000000000 +0200
++++ interval-839a03e1bddbafab868fbceee59abe678e32a0f3/src/Missing/Coquelicot.v	2019-12-03 14:55:15.821078400 +0100
+@@ -1,10 +1,13 @@
+ From Coq Require Import Reals Psatz.
+ From Coquelicot Require Import Coquelicot.
+-From mathcomp.ssreflect Require Import ssreflect ssrfun ssrbool eqtype ssrnat seq fintype bigop.
++From mathcomp.ssreflect Require Import ssreflect ssrfun ssrbool eqtype seq fintype bigop.
+ 
+ Require Import Stdlib.
+ Require Import MathComp.
+ 
++(* This needs to be here to have the Delimit Scope N_scope with num active *)
++From mathcomp.ssreflect Require Import ssrnat.
++
+ Lemma continuous_Rinv x :
+   x <> 0 ->
+   continuous (fun x => / x) x.
+@@ -631,7 +634,7 @@
+ 
+ rewrite /=.
+ apply: (filterlim_ext (fun x => minus (exp (-(lam * a)) / lam) (exp (-(lam * x)) / lam))).
+-move => x;rewrite /minus plus_comm; congr plus. rewrite /opp /=; field; lra.
++move => x; rewrite /minus Hierarchy.plus_comm; congr plus. rewrite /opp /=; field; lra.
+ rewrite /opp /=; field; lra.
+ rewrite /minus.
+ apply: (filterlim_comp _ _ _ (fun x => opp (exp (-(lam * x)) / lam)) (fun x => plus (exp (- (lam * a)) / lam) x) (Rbar_locally p_infty) (locally (0)) (locally (exp (- (lam * a)) / lam))); last first.
+--- interval-839a03e1bddbafab868fbceee59abe678e32a0f3.orig/src/Poly/Datatypes.v	2019-07-20 09:48:54.000000000 +0200
++++ interval-839a03e1bddbafab868fbceee59abe678e32a0f3/src/Poly/Datatypes.v	2019-12-03 15:42:40.401690500 +0100
+@@ -22,7 +22,7 @@
+ 
+ From Coq Require Import ZArith Reals.
+ From Coquelicot Require Import Coquelicot.
+-From mathcomp.ssreflect Require Import ssreflect ssrfun ssrbool eqtype ssrnat seq fintype bigop.
++From mathcomp.ssreflect Require Import ssreflect ssrfun ssrbool eqtype seq fintype bigop.
+ From Flocq Require Import Core.
+ 
+ Require Import Stdlib.
+@@ -31,6 +31,8 @@
+ Require Import Xreal.
+ Require Import Basic_rec.
+ 
++From mathcomp.ssreflect Require Import ssrnat.
++
+ Set Implicit Arguments.
+ Unset Strict Implicit.
+ Unset Printing Implicit Defensive.
+@@ -256,7 +258,9 @@
+ Section PrecIsPropagated.
+ Variable u : U.
+ 
+-Definition add := map2 (C.add u) id.
++Locate id.
```

Comment:
That's probably something to remove before submitting the patch upstream.

Reply:
Indeed. There are other places to clean up. E.g. I think that the issues with ssrnat is best fixed by changing Interval's "Missing" file for Mathcomponents. Also I am looking for a way to undefined Notations which overlap ssreflect definitions - currently I give explicit names. Do you happen to know a way how to undefined notations?

I hope for a beta it is OK as is.

Reply:
Hi, there is actually already code for undefining notations but no syntax for it. I'll make a proposal hopefully today or tomorrow.

Reply:
> Hi, there is actually already code for undefining notations but no syntax for it. I'll make a proposal hopefully today or tomorrow.

It would be great if we could get this into 8.11 (non beta). I think many of the issues in Interval could then be solved by undefiining certain new notations, possibly in a SSreflect compatibility file, which possibly should be part of the standard library (undefined new notations 8.X). The import issue I think can be fixed at a central place in interval, so that only very few lines would have to be changed.

Btw.: The already reported bug that Print Visibility does an anomaly if SSreflect is loaded doesn't make debugging such issues easier - it would be good to fix this in 8.11.

Reply:
There is a fix for the `Print Visibility` bug at #11276.

I have a branch for deactivating notations but it is not fully clear what user API we want for it.

For instance, assuming we have commands of the form:
```
Deactivate "_ + _" : nat_scope.
Reactivate "_ + _" : nat_scope.
Deactivate pred_of_simpl.
Reactivate pred_of_simpl.
```
Do we want them to have an effect at require-time of the .vo file in which they occur? Or do we want to consider them only as "interactive" features with an effect only over parsing/printing in the current document.

Reply:
The main purpose for disabling notations would be having a way to fix compatibility issues by hiding names in e.g. ssreflect with new notations from the standard library. This would only work in a comfortable way if the deactivation of standard library notations could be done by importing ssreflect.